src/test/java/seedu/sudohr/storage/JsonSerializableSudoHrTest.java

```diff
@@ -135,6 +137,13 @@ public void toModelType_nonExistentEmployeeInDepartment_throwsEmployeeNotFoundEx
         assertThrows(EmployeeNotFoundException.class, dataFromFile::toModelType);
     }
 
+    @Test
+    public void toModelType_inconsistentEmployeeInDepartment_throwsEmployeeNotFoundException() throws Exception {
```

Comment:
In my opinion, relying on memory address of an object couples everything too tightly. However, I honestly can't think of a concrete example where this coupling is a hindrance. We can discuss this in the future if needed, but I genuinely don't think it's an issue and makes everything clearer, because playing with a single reference might risk leading to unintended side effects imo.

Another argument is that current implementation has "final" fields. There's not much advantage to using single reference as you have to replace them anyways.

Reply:
okay i think your point is valid + what we discussed on tg. We'll use this as justification.